src/HomeScreen/hooks/useDatabaseUpdateCheck.js

```diff
@@ -0,0 +1,17 @@
+import { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { actions, checkForBaniDBUpdate } from "@common";
+
+const useDatabaseUpdateCheck = () => {
+  const dispatch = useDispatch();
+  const dbCheck = async () => {
+    const isUpdateAvailable = await checkForBaniDBUpdate();
+    dispatch(actions.toggleDatabaseUpdateAvailable(isUpdateAvailable));
+  };
+
+  useEffect(() => {
+    dbCheck();
```

Comment:
Since dbCheck is async, consider handling errors here to avoid silent failure.

```javascript
useEffect(() => {
  dbCheck().catch((err) => console.error("DB update check failed:", err));
}, [dbCheck]);
```